api-controllers/ServiceFabrikApiController.js

```diff
@@ -433,33 +369,39 @@ class ServiceFabrikApiController extends FabrikBaseController {
   abortLastBackup(req, res) {
     req.manager.verifyFeatureSupport('backup');
     const backupStartedAt = new Date();
-    return eventmesh.apiServerClient.getLastOperation({
-      resourceId: req.params.instance_id,
-      operationName: CONST.OPERATION_TYPE.BACKUP,
-      operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-    }).then(backupGuid => {
-      return eventmesh.apiServerClient.getOperationState({
+    return eventmesh
+      .apiServerClient.getLastOperation({
+        resourceId: req.params.instance_id,
         operationName: CONST.OPERATION_TYPE.BACKUP,
         operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-        operationId: backupGuid,
-      }).then(state => {
-        // abort only if the state is in progress
-        if (state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS) {
-          return eventmesh.apiServerClient.updateOperationState({
+      })
+      .then(backupGuid => {
+        return eventmesh
+          .apiServerClient.getOperationState({
             operationName: CONST.OPERATION_TYPE.BACKUP,
             operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
             operationId: backupGuid,
-            stateValue: CONST.OPERATION.ABORT
-          });
-        } else {
-          logger.info(`Skipping abort as state is : ${state}`);
-        }
-      }).then(() => ServiceFabrikApiController.getResourceOperationStatus({
-        operationId: backupGuid,
-        start_state: CONST.OPERATION.ABORT,
-        started_at: backupStartedAt
-      }));
-    }).then(status => res.status(status.state === 'aborting' ? CONST.HTTP_STATUS_CODE.ACCEPTED : CONST.HTTP_STATUS_CODE.OK).send({}));
+          })
+          .then(state => {
+            // abort only if the state is in progress
+            if (state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS) {
+              return eventmesh.apiServerClient.updateOperationState({
+                operationName: CONST.OPERATION_TYPE.BACKUP,
+                operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                operationId: backupGuid,
+                stateValue: CONST.OPERATION.ABORT
+              });
+            } else {
+              logger.info(`Skipping abort as state is : ${state}`);
+            }
+          })
+          .then(() => eventmesh.apiServerClient.getResourceOperationStatus({
```

Comment:
here dont you have to poll wait for the status to change from `ABORT` to `ABORTING` like you do at other places where you wait for `in-progress`?

Reply:
Aborting in sf v1 has been fire and forget . We had retained the same here.